src/tasks/start/backend/start-ganache.ts

```diff
@@ -27,8 +30,11 @@ export async function startGanache(
     )
   }
 
+  const nodeUrl = (bre.network.config as HttpNetworkConfig).url
+  const port = nodeUrl ? parseInt(new URL(nodeUrl).port) : testnetPort
```

Comment:
After discussing with @dapplion he told me this line is a bit problematic, if you set the url to `localhost:8545` it will throw.

>Using Node's built-in module for parsing URL has the "problem" / feature or requiring the protocol otherwise it throws.

Reply:
True. But the buidler also doesn't support such urls:

```
TypeError: Only HTTP(S) protocols are supported
    at getNodeRequestOptions (./node_modules/@nomiclabs/buidler/node_modules/node-fetch/lib/index.js:1303:9)
    at ./node_modules/@nomiclabs/buidler/node_modules/node-fetch/lib/index.js:1404:19
    at new Promise (<anonymous>)
    at fetch (./node_modules/@nomiclabs/buidler/node_modules/node-fetch/lib/index.js:1401:9)
    at HttpProvider._fetchJsonRpcResponse (./node_modules/@nomiclabs/buidler/src/internal/core/providers/http.ts:58:30)
```
Because under the hood it uses [node-fetch](https://github.com/nomiclabs/buidler/blob/v1.2.0/packages/buidler-core/src/internal/core/providers/http.ts#L58) which in turn relies on node.js core [url](https://github.com/node-fetch/node-fetch/blob/v2.6.0/src/request.js#L65) module which also expects protocol to be present:

```js
> url.parse('localhost:8545')
Url {
  protocol: 'localhost:',
  ...
```

I think we should add validation here and throw a similar error, something like `TypeError: Only HTTP protocol is supported`. We might as well require `hostname` to be `localhost`.

Reply:
@xuhcc That's a good point, then we should be okay by using `URL`